politeiad/cache/cockroachdb/decred.go

```diff
@@ -438,55 +447,62 @@ func (d *decred) voteResultsInsertRunoff(rfpToken string) error {
 }
 
 func (d *decred) voteResultsLoad(bestBlock uint64) error {
-	standard, runoff, err := d.voteResultsMissing(bestBlock)
-	if err != nil {
-		return err
-	}
-
-	// Insert vote results for standard votes proposals.
-	for _, token := range standard {
-		err := d.voteResultsInsertStandard(token)
+	// Check to see if the VoteResults table needs to be updated. Only update
+	// if best block is different from the last update.
+	d.bbMtx.RLock()
+	lbb := d.lastBestBlock
+	d.bbMtx.RUnlock()
+	if lbb != bestBlock {
```

Comment:
Can't you run this check a single time in `voteResultsMissing` and just return empty slices if its already been run for the provided best block.

Reply:
You're right mate, much cleaner code now.